docs/architecture/airbyte-specification.md

```diff
@@ -0,0 +1,180 @@
+# The Airbyte Specification
+
+## Summary
+This page will describe the Airbyte Specification.
+
+####Contents:
+1. [General information about the specification](#General)
+1. [Integration primitives](#Primitives)
+1. [Details of the protocol to pass information between integrations](#The-Protocol)
+
+## General
+* All structs described in this article are defined using JsonSchema.
+* Airbyte uses json serialized representations of these structs for all inter-process communication.
+
+### Definitions
+* Airbyte Worker - This is a core piece of the Airbyte stack that is responsible for initializing Sources and Destinations and passing data from Source to Destination. Someone implementing an integration need not ever touch this code, but in this article we mention it to contextualize how data is flowing through Airbyte.
+* Integration - An integration is code that allows Airbyte to interact with a specific underlying data source (e.g. Postgres). In Airbyte, an integration is either a Source or a Destination.
+* Source - An integration that _pulls_ data from an underlying data source. (e.g. A Postgres Source reads data from a Postgres database. A Stripe Source reads data from the Stripe API)
+* Destination - An integration that _pushes_ data to an underlying data source. (e.g. A Postgres Destination writes data to a Postgres database)
+* AirbyteSpecification - the specification that describe how to implement integrations using a standard interface.
+* AirbyteProtocol - the protocol used for inter-process communication.
+* Integration Commands - the commands that an integration container implements (e.g. spec, check, discover, read/write). We describe these commands in more detail below.
+* Sync - the act of moving data from a source to a destination.
+
+## Primitives
+
+### Source
+* A source is implemented as a docker container. The container must adhere to the following interface.
+
+Pseudocode of the interface:
+```
+spec() -> ConnectorSpecification
+check(Config) -> AirbyteConnectionStatus
+discover(Config) -> AirbyteCatalog
+read(Config, AirbyteCatalog, State) -> Stream<AirbyteMessage>
+```
+
+How to the container will be called:
+The first argument passed to the image must be the command (e.g. spec, check, discover, read). Additional arguments can be passed after the command.
+Note: When called by the airbyte worker, it will handle mounting the appropriate paths so that the config files are available to the container. This code snippet does not include that logic.
```

Comment:
```suggestion
Note: The system running the container will handle mounting the appropriate paths so that the config files are available to the container. This code snippet does not include that logic.
```